pdm/cli/commands/run.py

```diff
@@ -119,23 +123,31 @@ def _run_process(
         shell: bool = False,
         site_packages: bool = False,
         env: Mapping[str, str] | None = None,
-        env_file: str | None = None,
+        env_file: EnvFileOptions | str | None = None,
     ) -> int:
         """Run command in a subprocess and return the exit code."""
         project = self.project
-        process_env = {}
-        if env_file:
+        process_env = os.environ.copy()
+        if env_file is not None:
+            if isinstance(env_file, str):
+                path = env_file
+                override = False
+            else:
+                path = env_file["override"]
+                override = True
+
             import dotenv
 
             project.core.ui.echo(
                 f"Loading .env file: [green]{env_file}[/]",
                 err=True,
                 verbosity=termui.Verbosity.DETAIL,
             )
-            process_env = dotenv.dotenv_values(
-                project.root / env_file, encoding="utf-8"
-            )
-        process_env.update(os.environ)
+            dotenv_env = dotenv.dotenv_values(project.root / path, encoding="utf-8")
+            if override:
+                process_env = process_env | dotenv_env
```

Comment:
This syntax is Python 3.9+, please use `{**this, **other}` instead.

Reply:
Done.